src/include/txn.i

```diff
@@ -786,11 +835,22 @@ static inline int
 __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 {
 	WT_TXN *txn;
+	uint32_t ignore_prepare_set;
 
 	txn = &session->txn;
+	ignore_prepare_set = F_ISSET(txn, WT_TXN_IGNORE_PREPARE);
+	/*
+	 * Clear the ignore prepare setting of txn, as it is not suppose to
+	 * affect the visibility for update operation
+	 */
+	if (ignore_prepare_set)
+		F_CLR(txn, WT_TXN_IGNORE_PREPARE);
+
 	if (txn->isolation == WT_ISO_SNAPSHOT)
 		while (upd != NULL && !__wt_txn_upd_visible(session, upd)) {
 			if (upd->txnid != WT_TXN_ABORTED) {
+				if (ignore_prepare_set)
+					F_SET(txn, WT_TXN_IGNORE_PREPARE);
 				WT_STAT_CONN_INCR(
 				    session, txn_update_conflict);
 				WT_STAT_DATA_INCR(
```

Comment:
The `WT_STAT_CONN_INCR` and `WT_STAT_DATA_INCR` lines can be joined, they don't need to be split.